pkg/kv/kvpb/api.proto

```diff
@@ -2200,6 +2165,53 @@ message AddSSTableResponse {
   bytes following_likely_non_empty_span_start = 4 [(gogoproto.casttype) = "github.com/cockroachdb/cockroach/pkg/roachpb.Key"];
 }
 
+// LinkExternalSSTableRequest contains arguments to the LinkExternalSSTable
+// method, which links an External SST file into the Pebble log-structured
+// merge-tree. The External SST must point to a backup file which contains
+// versioned values with non-zero MVCC timestamps (no intents or inline values).
+// It cannot be used in a transaction, cannot be split across ranges, and must
+// be alone in a batch.
+message LinkExternalSSTableRequest {
+    RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+
+  // SSTTimestampToRequestTimestamp gives the timestamp used for all MVCC keys
+  // in the provided SST. If this timestamp differs from the request timestamp
+  // (e.g. if the request gets pushed) then all MVCC keys in the SST will be
+  // rewritten to the request timestamp during request evaluation. This ensures
+  // the writes comply with the timestamp cache and closed timestamp. It also
+  // causes the External SSTable to be emitted via the range feed, since it
+  // respects the closed timestamp.
+  //
+  // Callers should always set this, except in very special circumstances when
+  // the timestamp cache and closed timestamp can safely be ignored (e.g.
+  // restore of a tenant).
+  //
+  // Note that this alone is not sufficient to guarantee MVCC correctness, since
+  // it can write below or replace existing committed versions (the tscache is
+  // only bumped when the values are subsequently read).
+  util.hlc.Timestamp sst_timestamp_to_request_timestamp = 2 [
+    (gogoproto.customname) = "SSTTimestampToRequestTimestamp", (gogoproto.nullable) = false];
+
+  // MVCCStats are estimated MVCCStats for the contents of this Extrenal SSTable
+  // and is used as-is during evaluation of the command to update the range
+  // MVCCStats, instead of computing the stats for the SSTable by iterating it.
+  storage.enginepb.MVCCStats mvcc_stats = 3 [(gogoproto.customname) = "MVCCStats"];
```

Comment:
Maybe move this into ExternalFile so if/when we repeat it, the stats are per-file, not per-request.